misc/helm-charts/operator/README.md.gotmpl

```diff
@@ -202,6 +202,106 @@ kubectl logs -l app.kubernetes.io/name=materialize-operator -n materialize
 
 For more detailed information on using and troubleshooting the Materialize operator, refer to the [Materialize documentation](https://materialize.com/docs).
 
+## Upgrading Materialize
+
+Once you have the Materialize operator installed, with the operator managing your Materialize environments, you can upgrade the operator and the environments independently.
+
+### Upgrading the Helm Chart
+
+To upgrade the Materialize operator to a new version:
+
+```shell
+helm upgrade my-materialize-operator materialize/misc/helm-charts/operator
+```
+
+If you have custom values, make sure to include your values file:
+
+```shell
+helm upgrade my-materialize-operator materialize/misc/helm-charts/operator -f my-values.yaml
+```
+
+### Upgrading Materialize Environments
+
+To upgrade your Materialize environments, you'll need to update the Materialize custom resource and trigger a rollout.
+
+Note, that this will trigger a zero-downtime rolling upgrade of the Materialize environment, meaning that you will have two versions of the environment running simultaneously during the upgrade. Make sure that you have enough resources on your Kubernetes cluster to support this.
+
+#### Using `kubectl` patch
+
+For normal upgrades when there are changes to the environment (like image updates):
+
+```shell
+# Generate a new UUID and patch the Materialize environment
+kubectl patch materialize <environment-name> \
```

Comment:
wouldn't mind an explainer on what version `Materialize` gets bumped to if this is used